tests/unit/src/main/scala/tests/TestingServer.scala

```diff
@@ -393,16 +393,33 @@ final class TestingServer(
     textDocumentCapabilities.setFoldingRange(new FoldingRangeCapabilities)
     val initOptions = initializationOptions.getOrElse(
       InitializationOptions.Default.copy(
-        debuggingProvider = true,
-        treeViewProvider = true,
-        slowTaskProvider = true
+        debuggingProvider = Some(true),
+        treeViewProvider = Some(true),
+        slowTaskProvider = Some(true)
       )
     )
+
+    // Yes, this is a bit gross :/
+    // However, I want to only get the existing fields that are being set
+    // much like it'd be when a client actually sends this. This will just
+    // collect the fields that are set, get the values, and then make them into
+    // a map that will become a JsonObject to pass in as the InitializationOptions
+    val existingInitOptions =
+      initOptions.getClass.getDeclaredFields
+        .foldLeft(Map.empty[String, Any]) { (acc, field) =>
+          field.setAccessible(true)
+          acc + (field.getName -> field.get(initOptions))
+        }
+        .collect {
+          case (key, Some(value)) => key -> value
+        }
+        .asJava
```

Comment:
So I'm open to some feedback here about how we should do this. The tests starting failing after the changes due to us just calling `.toJson` here, which would convert our `Option`s to `{}`. That would cause the `InitializationOptions` to just be default not matter what. I thought about writing something for gson that would turn an Option into the value it holds, but then for `None` we are stuck with null. Since most clients aren't just going to send a valid key with a `null` field for this, I wanted to actually mimic how they'd do it by _only_ sending in the keys that they would want to set. So in the end, this does just that and sends in an object only with the desired keys set. I think this is actually good for testing as it more resembles what happens in real life, but then I'm stuck with this gross thing because of how we build up the `InitializtionOptions` in the test and then convert them to json. Thoughts?